src/Microsoft.Azure.CosmosRepository/InMemoryRepository.cs

```diff
@@ -346,6 +349,29 @@ public async ValueTask<int> CountAsync(CancellationToken cancellationToken = def
                 0);
         }
 
+        public async ValueTask<IPageQueryResult<TItem>> PageAsync(ISpecification<TItem> specification, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+
+            IQueryable<TItem> query = Items.Values.Select(DeserializeItem).AsQueryable()
+                .Where(item => item.Type == typeof(TItem).Name);
+
+            int pageSize = specification.PageSize;
+            ISpecificationEvaluator specificationEvaluator = new SpecificationEvaluator();
```

Comment:
Can this be injected via DI?

Reply:
Is there a pattern for DI use in the InMemoryRepository? 
It's already using DI in defaultrepository

Reply:
The `InMemoryRepository` technically supports full DI, as it's added to the service collection/provider. You should be able to add it as a .ctor parameter, and DI would provide it.